jetty9-base/src/main/java/com/google/cloud/runtimes/jetty9/TracingLogHandler.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.jetty9;
+
+import java.util.logging.ConsoleHandler;
+import java.util.logging.LogRecord;
+
+/**
+ * A Logging Handler extended with a request traceId label.
+ */
+public class TracingLogHandler extends ConsoleHandler {
+
+  /**
+   * Construct a TracingLogHandler.
+   */
+  public TracingLogHandler() {}
+
+  @Override
+  public synchronized void publish(LogRecord record) {
+    String traceId = RequestContextScope.getCurrentTraceId();
+    if (traceId != null) {
+      record.setMessage(record.getMessage() + " traceId=" + traceId);
```

Comment:
1) Yes, we should add traceId as a log record param and provide a special formatter.
2) I'd defer the decision on the ultimate location of this code to later. For now, let's keep it in this repo.

I don't think this is really a step away from using Stackdriver as a log sink. If you're sending logs to Stackdriver via fluentd, then the corresponding log formatter would make trace id part of the message, whereas if you're sending log entry directly via RPC, the corresponding subclass of the `TracingLogHandler` wouldn't need a formatter at all, but could still use the log record param.